pkg/controllers/gateway/gateway_controller.go

```diff
@@ -207,13 +201,14 @@ func (r *GatewayReconciler) reconcileDownstreamFromUpstreamGateway(ctx context.C
 		downstream.Labels = map[string]string{}
 	}
 	downstream.Labels[ManagedLabel] = "true"
-	accessor := traffic.NewGateway(downstream)
-	if isDeleting(downstream) {
+	accessor := traffic.NewGateway(upstreamGateway)
+	if isDeleting(upstreamGateway) {
```

Comment:
How does it get here? Do we not return earlier if this is deleting?

Reply:
it calls the reconcile in the deleting block

Reply:
https://github.com/Kuadrant/multicluster-gateway-controller/pull/216/files#diff-6afa7d73841a5c0ebdb78eaf859ce5a400203a037f871440940b5347c3a97376R125

Reply:
ignore, i read this as being further down the Reconcile function